### ISSUE TITLE
Connection problems / Couldn't find a suitable mixer

### ISSUE THREAD

**Issue by paulbastian** [librespot-org/librespot-java#220]
Hello,

I'm trying to make this software work on my Raspberry Pi2+Hifiberry Soundboard but unfortunatly I'm stuck. I start the latest release of librespot-core-jar-with-dependencies.jar and I get an error all the time indicating a problem with the mixer, although I'm not sure from if this is the real problem reading from other issue threads here. I am successfully running the rust spotifyd build, but running into the ominous "code 104 connection reset by peer" issue there.

So however the sound is working with aplay as well, I tried some of the suggestions here in the forum, changing Java SDK and using the sound.properties file, installed pulseaudio as well. Any ideas?
Note the timestamps after the first stack trace about the mixer, it hangs for over a minute.
Thanks very much!

Log:

```
> 2020-05-02 19:08:45,657 [main] INFO  Utils:73 - Cryptography restrictions removal not needed.
> 2020-05-02 19:08:46,404 [main] INFO  ApResolver:66 - Loaded aps into pool: {accesspoint=[gew1-accesspoint-b-82wn.ap.spotify.com:4070, gew1-accesspoint-b-jb5p.ap.spotify.com:443, gew1-accesspoint-b-czzt.ap.spotify.com:80, gew1-accesspoint-b-fkgc.ap.spotify.com:4070, gew1-accesspoint-b-4plq.ap.spotify.com:443, gew1-accesspoint-b-hbkf.ap.spotify.com:80, guc3-accesspoint-b-mh02.ap.spotify.com:4070, guc3-accesspoint-b-8gfw.ap.spotify.com:443, guc3-accesspoint-b-vks2.ap.spotify.com:80], dealer=[gew-dealer.spotify.com:443, gae-dealer.spotify.com:443, guc-dealer.spotify.com:443], spclient=[gew-spclient.spotify.com:443, guc-spclient.spotify.com:443, gae-spclient.spotify.com:443]}
> 2020-05-02 19:08:48,153 [main] INFO  ZeroconfServer:383 - Zeroconf HTTP server started successfully on port 51938!
> 2020-05-02 19:08:49,453 [zeroconf-client-2252113] TRACE ZeroconfServer:447 - Handling request: GET /?action=getInfo HTTP/1.1, headers: {Keep-Alive=0, User-Agent=Spotify/113100703 Win32/0 (PC laptop), Connection=keep-alive, Host=192.168.2.201, Accept-Encoding=gzip, Content-Type=application/x-www-form-urlencoded}
> 2020-05-02 19:08:49,736 [zeroconf-client-7802905] TRACE ZeroconfServer:447 - Handling request: GET /?action=getInfo HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/8.5.55 Android/29 (SM-G965F), Host=192.168.2.201, Accept-Encoding=gzip, Content-Type=application/x-www-form-urlencoded}
> 2020-05-02 19:08:49,813 [main] INFO  Zeroconf:456 - Announced service: Service{alias='librespot-java', service='spotify-connect', port=51938, text={VERSION=1.0, CPath=/, Stack=SP}, addresses=[/192.168.2.201], domain='.local', protocol='tcp', host='soundpi1'}
> 2020-05-02 19:08:55,258 [zeroconf-client-2252113] TRACE ZeroconfServer:447 - Handling request: GET /?action=getInfo&version=2.7.1 HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/8.5.55 Android/29 (SM-G965F), Host=192.168.2.201, Accept-Encoding=gzip, Content-Type=application/x-www-form-urlencoded}
> 2020-05-02 19:08:56,667 [zeroconf-client-2252113] TRACE ZeroconfServer:447 - Handling request: GET /?action=getInfo&version=2.7.1 HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/8.5.55 Android/29 (SM-G965F), Host=192.168.2.201, Accept-Encoding=gzip, Content-Type=application/x-www-form-urlencoded}
> 2020-05-02 19:09:01,247 [zeroconf-client-2252113] TRACE ZeroconfServer:447 - Handling request: POST / HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/8.5.55 Android/29 (SM-G965F), Host=192.168.2.201, Accept-Encoding=gzip, Content-Length=753, Content-Type=application/x-www-form-urlencoded}
> 2020-05-02 19:09:05,392 [zeroconf-client-2252113] DEBUG TimeProvider:73 - Loaded time offset from NTP: 0ms
> 2020-05-02 19:09:12,022 [zeroconf-client-2252113] INFO  Session:122 - Created new session! {deviceId: 2651de24cf6a406b3371334e42f29b479c704aa4, ap: gew1-accesspoint-b-4plq.ap.spotify.com:443, proxy: false} 
> 2020-05-02 19:09:12,027 [zeroconf-client-2252113] INFO  ZeroconfServer:343 - Accepted new user from SM-G965F. {deviceId: 2651de24cf6a406b3371334e42f29b479c704aa4}
> 2020-05-02 19:09:14,778 [zeroconf-client-2252113] INFO  Session:320 - Connected successfully!
> 2020-05-02 19:09:15,124 [session-packet-receiver] TRACE Session:1073 - Session.Receiver started
> 2020-05-02 19:09:15,206 [zeroconf-client-2252113] TRACE AsyncProcessor:33 - AsyncProcessor{pm-mercury} has started
> 2020-05-02 19:09:15,228 [zeroconf-client-2252113] TRACE AsyncProcessor:33 - AsyncProcessor{pm-audio-keys} has started
> 2020-05-02 19:09:15,288 [zeroconf-client-2252113] TRACE AsyncProcessor:33 - AsyncProcessor{pm-channels} has started
> 2020-05-02 19:09:15,403 [cache-maintenance] INFO  CacheManager:71 - There are 0 cached entries.
> 2020-05-02 19:09:15,429 [zeroconf-client-2252113] TRACE AsyncProcessor:33 - AsyncProcessor{dealer-worker} has started
> Exception in thread "zeroconf-client-2252113" xyz.gianlu.librespot.player.mixing.LineHelper$MixerException: Couldn't find a suitable mixer, openLine: interface SourceDataLine supporting format PCM_SIGNED 44100.0 Hz, 16 bit, stereo, 4 bytes/frame, little-endian, available: []
> 	at xyz.gianlu.librespot.player.mixing.LineHelper.findSupportingMixersFor(LineHelper.java:33)
> 	at xyz.gianlu.librespot.player.mixing.LineHelper.getLineFor(LineHelper.java:60)
> 	at xyz.gianlu.librespot.player.PlayerRunner$Output.acquireLine(PlayerRunner.java:408)
> 	at xyz.gianlu.librespot.player.PlayerRunner$Output.<init>(PlayerRunner.java:388)
> 	at xyz.gianlu.librespot.player.PlayerRunner.<init>(PlayerRunner.java:69)
> 	at xyz.gianlu.librespot.player.Player.<init>(Player.java:62)
> 	at xyz.gianlu.librespot.core.Session.authenticate(Session.java:340)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.handleAddUser(ZeroconfServer.java:346)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$100(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handleRequest(ZeroconfServer.java:410)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:485)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> 2020-05-02 19:11:20,159 [session-scheduler-16844628] WARN  Session:1100 - Socket timed out. Reconnecting...
> Exception in thread "session-packet-receiver" java.lang.IllegalStateException: java.lang.InterruptedException
> 	at xyz.gianlu.librespot.core.Session.send(Session.java:509)
> 	at xyz.gianlu.librespot.core.Session$Receiver.run(Session.java:1107)
> 	at java.lang.Thread.run(Thread.java:748)
> Caused by: java.lang.InterruptedException
> 	at java.lang.Object.wait(Native Method)
> 	at java.lang.Object.wait(Object.java:502)
> 	at xyz.gianlu.librespot.core.Session.send(Session.java:507)
> 	... 2 more
> 2020-05-02 19:11:22,623 [session-scheduler-16844628] INFO  Session:320 - Connected successfully!
> 2020-05-02 19:11:22,901 [session-packet-receiver] TRACE Session:1073 - Session.Receiver started
> Exception in thread "zeroconf-client-29536826" 2020-05-02 19:11:22,911 [session-scheduler-16844628] INFO  Session:665 - Re-authenticated as wamvxztrfpcomxphv6401as55!
> java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 2020-05-02 19:11:22,918 [session-packet-receiver] INFO  Session:1155 - Skipping SecretBlock
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> 2020-05-02 19:11:22,923 [session-packet-receiver] INFO  Session:1127 - Received LicenseVersion: 0
> Exception in thread "zeroconf-client-18880404" java.lang.IllegalStateException: Session isn't authenticated!2020-05-02 19:11:22,928 [session-packet-receiver] INFO  Session:1117 - Received CountryCode: DE
> 
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> 2020-05-02 19:11:23,401 [session-packet-receiver] TRACE Session:716 - Parsed product info: {remote-control=6, audio-preview-url-template=https://p.scdn.co/mp3-preview/{id}, incognito_mode_timeout=21600, ad-formats-preroll-video=0, type=premium, ab_recently_played_feature_time_filter_threshold=com.spotify.gaia=30,driving-mode=120,spotify%3Ainternal%3Astartpage=30, license-agreements=, publish-playlist=1, buffering-strategy=0, expiry=1, ab-desktop-hide-follow=0, ab-collection-hide-unavailable-albums=0, capper-profile=, payments-initial-campaign=default, profile-image-upload=1, enable-annotations=2, ab-ad-player-targeting=1, india-experience=0, video-wifi-initial-bitrate=800000, ab-mobile-running-onlymanualmode=only-manual, playlist-annotations-markup=0, prefetch-keys=1, allow-override-internal-prefs=0, prefetch-window-max=2, ab-mandatory-trials=0, send-email=0, ad-formats-video-takeover=1, npt-disabled=2, ab-collection-bookmark-model=1, pause-after=0, local-files-import=0, video-manifest-url=https://spclient.wg.spotify.com/manifests/v6/{type}/sources/{source_id}/options/supports_drm, ab-sugarpills-sanity-check=0, playlist-folders=1, arsenal_country=1, track-cap=0, ab-mobile-running-tempo-detection=Control, shows-collection=1, enable-annotations-read=0, mobile-browse=0, payments-locked-state=0, shows-collection-jam=1, ab-browse-music-tuesday=1, offline=1, streaming=1, fb-info-confirmation=control, taste-onboarding-disabled=0, wanted-licenses=, lastfm-session=, ab-desktop-playlist-annotation-edit=1, ab-mobile-discover=0, widevine-license-url=https://spclient.wg.spotify.com/widevine-license/v1/video/license, key-caching-max-count=5000, ad-session-persistence=1, video-initial-bitrate=200000, ab-play-history=0, created_by_partner=, mobile-payment=0, payment-state=, key-caching-max-offline-seconds=1800, ugc-abuse-report-url=https://support.spotify.com/abuse/?uri={uri}, shuffle-algorithm=1, use-pl3=0, image-url=https://i.scdn.co/image/{file_id}, use-playlist-app=0, enable-autostart=1, metadata-link-lookup-modes=0, shuffle=0, instant-search=0, instant-search-expand-sidebar=0, use-playlist-uris=0, user-profile-show-invitation-codes=0, ab-ad-requester=1, nft-disabled=1, payments-cancel-state-interstitial=0, ab-collection-union=1, key-memory-cache-mode=1:15,300, libspotify=1, high-bitrate=1, head-file-caching=1, radio=1, explicit-content=1, video-cdn-sampling=1, ab-android-push-notifications=1, unrestricted=1, use-fb-publish-backend=2, app-developer=0, buffering-strategy-parameters=0.8:0.2:0.0:0.0:0.0:0.0:1.0:10:10:2000:10000:10485760, enable-gapless=1, ab-playlist-extender=5, enable-crossfade=1, ad-persist-reward-time=0, network-operator-premium-activation=1, video-device-blacklisted=0, collection=1, fb-grant-permission-local-render=0, backend-advised-bitrate=1, catalogue=premium, storage-size-config=10240,90,500,3, ads=0, head-files-url=https://heads4-ak-spotify-com.akamaized.net/head/{file_id}, ab-collection-followed-artists-only=0, browse-overview-enabled=1, on-demand=1, ap-resolve-pods=0, name=Spotify Premium, sidebar-navigation-enabled=0, loudness-levels=1:-5.0,3.0,6.0:-2.0, mobile-login=1, preferred-locale=en, license-acceptance-grace-days=30, prefetch-strategy=2, ugc-abuse-report=1, ab-watch-now=0, language-onboarding=0, financial-product=pr:premium,tc:0,rt:v2_DE_default_new-family-sub_0_EUR_default, ab-collection-offline-mode=0, streaming-rules=, head-files=1, capping-bar-threshold=3601, ab-new-share-flow=0, video-keyframe-url=http://keyframes-fa.cdn.spotify.com/keyframes/v1/sources/{source_id}/keyframe/heights/{height}/timestamps/{timestamp_ms}.jpg, mobile=1, ab-mobile-startpage=0, ab-hub-artist=1, ab-moments-experience=0, ab-mobile-social-feed=1, ab-test-group=366, player-license=premium}
> 2020-05-02 19:11:23,408 [session-packet-receiver] INFO  Session:1155 - Skipping LegacyWelcome
> 2020-05-02 19:11:23,411 [session-packet-receiver] INFO  Session:1155 - Skipping UnknownData_AllZeros
> 2020-05-02 19:11:23,437 [session-packet-receiver] DEBUG Session:1131 - Received 0x10: BAB84BCFB6DE059C018CAA71E0C3430C6A0AD15C
> 2020-05-02 19:11:23,445 [session-packet-receiver] INFO  Session:1082 - Skipping unknown command {cmd: 0x75, payload: 000000}
> 2020-05-02 19:11:23,465 [handle-packet-3874367] TRACE MercuryClient:176 - Handling packet, cmd: MercuryEvent, seq: -3846888819890585600, flags: 1, parts: 1
> 2020-05-02 19:11:23,523 [handle-packet-3874367] DEBUG MercuryClient:209 - Couldn't dispatch Mercury event {seq: -3846888819890585600, uri: hm://pusher/v1/connections/MjY1MWRlMjRjZjZhNDA2YjMzNzEzMzRlNDJmMjliNDc5YzcwNGFhNCtBUCt0Y3A6Ly9ndWMzLWFjY2Vzc3BvaW50LWItdmtzMi5ndWMzLnNwb3RpZnkubmV0OjUwMjgrNDgwRkY4RjMxODg3NDFDNzBDMjY3Q0EyQjU4RDg0RjZBOTYyRkJEQzVDNTcwQkJEMTMyOUVCMTBFMEE1RUExOA%3D%3D, code: 200, payload: []}
> Exception in thread "zeroconf-client-22868262" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Exception in thread "zeroconf-client-27091377" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Exception in thread "zeroconf-client-21434752" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Exception in thread "zeroconf-client-24632236" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Exception in thread "zeroconf-client-7090549" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Exception in thread "zeroconf-client-8233429" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Exception in thread "zeroconf-client-893470" java.lang.IllegalStateException: Session isn't authenticated!
> 	at xyz.gianlu.librespot.core.Session.player(Session.java:583)
> 	at xyz.gianlu.librespot.core.Session.isActive(Session.java:607)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.hasActiveSession(ZeroconfServer.java:235)
> 	at xyz.gianlu.librespot.core.ZeroconfServer.access$300(ZeroconfServer.java:39)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.handle(ZeroconfServer.java:446)
> 	at xyz.gianlu.librespot.core.ZeroconfServer$HttpRunner.lambda$run$1(ZeroconfServer.java:393)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> 
```

**Comment by devgianlu** [comment]
This is seems purely a configuration issue, the application cannot see any mixer. Running the program as `root` should solve the issue, you can also search on the internet how to make your user see mixers.

**Comment by paulbastian** [comment]
running as root does not solve the problem unfortunately. any other ideas?

**Comment by devgianlu** [comment]
Have you tried installing `alsa`?

**Comment by paulbastian** [comment]
yes alsa is installed of course

**Comment by devgianlu** [comment]
@spocon Any idea?

**Comment by spocon** [comment]
@paulbastian  
- which java version do you run it ? openjdk or oracle ? ( java -version)
- did you run it with config changes ( --conf-file=config.toml)  at the end or the standard configurations from the jar file ?
- which OS are you using ?

the mixers are found by the java sdk if im not mistaken....

**Comment by paulbastian** [comment]
Thanks for you help guys!!
I have tried openjdk13 and openjdk8, i've tried the various adaptions in /jre/lib/sound.properties.
I'm using the default generated config.toml
I'm running arch linux on a raspberry pi2 with a hifiberry soundboard, alsa and pulseaudio installed
/etc/asound.conf = 
```
pcm.!default {
 type hw card 0
}
ctl.!default { 
 type hw card 0
}
```

**Comment by spocon** [comment]
I assume the problem is a config issue from Linux arch . Have you tried this ?

https://www.raspberrypi.org/forums/viewtopic.php?t=30830

**Comment by paulbastian** [comment]
Its not like I have trouble with playing sound in general, its just with this program or maybe java I presume

**Comment by spocon** [comment]
Could you send the output from:
`lsmod`

**Comment by paulbastian** [comment]
```
Module                  Size  Used by
snd_soc_pcm512x_i2c    16384  1
snd_soc_pcm512x        28672  1 snd_soc_pcm512x_i2c
bcm2835_codec          36864  0
bcm2835_v4l2           45056  0
bcm2835_isp            28672  0
v4l2_mem2mem           28672  1 bcm2835_codec
bcm2835_mmal_vchiq     28672  3 bcm2835_isp,bcm2835_codec,bcm2835_v4l2
videobuf2_dma_contig    20480  2 bcm2835_isp,bcm2835_codec
videobuf2_vmalloc      16384  1 bcm2835_v4l2
videobuf2_memops       16384  2 videobuf2_dma_contig,videobuf2_vmalloc
videobuf2_v4l2         24576  4 bcm2835_isp,bcm2835_codec,bcm2835_v4l2,v4l2_mem2mem
videobuf2_common       49152  5 bcm2835_isp,bcm2835_codec,bcm2835_v4l2,v4l2_mem2mem,videobuf2_v4l2
snd_soc_hifiberry_dacplus    16384  0
snd_soc_bcm2835_i2s    16384  2
videodev              212992  6 bcm2835_isp,bcm2835_codec,videobuf2_common,bcm2835_v4l2,v4l2_mem2mem,videobuf2_v4l2
raspberrypi_hwmon      16384  0
hwmon                  28672  1 raspberrypi_hwmon
snd_soc_core          188416  3 snd_soc_pcm512x,snd_soc_hifiberry_dacplus,snd_soc_bcm2835_i2s
i2c_bcm2835            16384  0
mc                     40960  6 bcm2835_isp,bcm2835_codec,videobuf2_common,videodev,v4l2_mem2mem,videobuf2_v4l2
snd_compress           20480  1 snd_soc_core
snd_pcm_dmaengine      16384  1 snd_soc_core
bcm2835_rng            16384  0
rng_core               16384  1 bcm2835_rng
uio_pdrv_genirq        16384  0
clk_hifiberry_dacpro    16384  1
uio                    20480  1 uio_pdrv_genirq
fixed                  16384  3
sch_fq_codel           20480  2
snd_bcm2835            24576  0
snd_pcm                94208  6 snd_soc_pcm512x,snd_pcm_dmaengine,snd_soc_hifiberry_dacplus,snd_soc_bcm2835_i2s,snd_bcm2835,snd_soc_core
snd_timer              32768  1 snd_pcm
snd                    69632  6 snd_compress,snd_soc_pcm512x,snd_timer,snd_bcm2835,snd_soc_core,snd_pcm
drm                   430080  0
drm_panel_orientation_quirks    16384  1 drm
ip_tables              28672  0
x_tables               32768  1 ip_tables
ipv6                  475136  44
nf_defrag_ipv6         16384  1 ipv6
```

**Comment by paulbastian** [comment]
Also I forget to mention I am using openjdk-headless instead of the standard one, but that should not be a problem? I've been troubleshooting for hours :(

**Comment by spocon** [comment]
First of all please disable the internal soundcard snd_bcm2835
 As described here:

https://github.com/guussie/PiDS/wiki/09.-How-to-make-various-DACs-work

Please print the out from this commands
`aplay -l`
and
`lsmod`

**Comment by paulbastian** [comment]
Parts of the guide aren't legit anymore because it was written for Linux 3.x and the link now even refer to Hifiberry's Linux 4.x guide.
aplay -l shows all good:
```
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_dacplus], device 0: HiFiBerry DAC+ HiFi pcm512x-hifi-0 [HiFiBerry DAC+ HiFi pcm512x-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

```

lsmod now:
```
Module                  Size  Used by
snd_soc_pcm512x_i2c    16384  1
snd_soc_pcm512x        28672  1 snd_soc_pcm512x_i2c
raspberrypi_hwmon      16384  0
hwmon                  28672  1 raspberrypi_hwmon
snd_soc_hifiberry_dacplus    16384  0
snd_soc_bcm2835_i2s    16384  2
snd_soc_core          188416  3 snd_soc_pcm512x,snd_soc_hifiberry_dacplus,snd_soc_bcm2835_i2s
i2c_bcm2835            16384  0
snd_compress           20480  1 snd_soc_core
snd_pcm_dmaengine      16384  1 snd_soc_core
snd_pcm                94208  5 snd_soc_pcm512x,snd_pcm_dmaengine,snd_soc_hifiberry_dacplus,snd_soc_bcm2835_i2s,snd_soc_core
snd_timer              32768  1 snd_pcm
bcm2835_codec          36864  0
snd                    69632  5 snd_compress,snd_soc_pcm512x,snd_timer,snd_soc_core,snd_pcm
bcm2835_v4l2           45056  0
v4l2_mem2mem           28672  1 bcm2835_codec
bcm2835_isp            28672  0
bcm2835_mmal_vchiq     28672  3 bcm2835_isp,bcm2835_codec,bcm2835_v4l2
videobuf2_vmalloc      16384  1 bcm2835_v4l2
videobuf2_dma_contig    20480  2 bcm2835_isp,bcm2835_codec
videobuf2_memops       16384  2 videobuf2_dma_contig,videobuf2_vmalloc
bcm2835_rng            16384  0
videobuf2_v4l2         24576  4 bcm2835_isp,bcm2835_codec,bcm2835_v4l2,v4l2_mem2mem
videobuf2_common       49152  5 bcm2835_isp,bcm2835_codec,bcm2835_v4l2,v4l2_mem2mem,videobuf2_v4l2
rng_core               16384  1 bcm2835_rng
videodev              212992  6 bcm2835_isp,bcm2835_codec,videobuf2_common,bcm2835_v4l2,v4l2_mem2mem,videobuf2_v4l2
mc                     40960  6 bcm2835_isp,bcm2835_codec,videobuf2_common,videodev,v4l2_mem2mem,videobuf2_v4l2
uio_pdrv_genirq        16384  0
clk_hifiberry_dacpro    16384  1
fixed                  16384  3
uio                    20480  1 uio_pdrv_genirq
sch_fq_codel           20480  2
drm                   430080  0
drm_panel_orientation_quirks    16384  1 drm
ip_tables              28672  0
x_tables               32768  1 ip_tables
ipv6                  475136  44
nf_defrag_ipv6         16384  1 ipv6
```

**Comment by spocon** [comment]
What happens when you put 
mixerkeywords: "snd_rpi_hifiberry_dacplus" 

https://github.com/dtcooper/raspotify/issues/58

**Comment by Kucheek** [comment]
> Also I forget to mention I am using openjdk-headless instead of the standard one, but that should not be a problem? I've been troubleshooting for hours :(

If you run out of ideas, you can always try openJDK headful. I have a similar setup, RPi 1B with Hifiberry DAC+, and when I first tried librespot-java, it also could not see my soundcard. Can't remember the exact cause, but I ended up with non-headless openjdk for some reason. There might be some additional libraries as certain articles indicate:

[https://www.raspberrypi.org/forums/viewtopic.php?t=15014](https://www.raspberrypi.org/forums/viewtopic.php?t=15014)

**Comment by paulbastian** [comment]
ok thanks guys, I got the next step.
Indeed the error is within jdk-headless, as it does not bring all necessary sound libraries.
However, I'm not finished in this quest yet :(
I got some diffrent behaviour now, eith no sound or stuttering sound, often seeing 100% CPU usage.
Hopefully this log might help, from what I see the program is running twice through the mixer finding process, is that correct?
```
2020-05-10 16:40:43,443 [main] INFO  Utils:73 - Cryptography restrictions removal not needed.
2020-05-10 16:40:44,194 [main] INFO  ApResolver:66 - Loaded aps into pool: {accesspoint=[gew1-accesspoint-b-91rz.ap.spotify.com:4070, gew1-accesspoint-b-7q0z.ap.spotify.com:443, gew1-accesspoint-b-s68v.ap.spotify.com:80, gew1-accesspoint-b-67v0.ap.spotify.com:4070, gew1-accesspoint-b-gf3s.ap.spotify.com:443, gew1-accesspoint-b-psb5.ap.spotify.com:80, guc3-accesspoint-b-4zq1.ap.spotify.com:4070, gae2-accesspoint-b-nfnq.ap.spotify.com:443, guc3-accesspoint-b-gxh8.ap.spotify.com:80], dealer=[gew-dealer.spotify.com:443, gae-dealer.spotify.com:443, guc-dealer.spotify.com:443], spclient=[gew-spclient.spotify.com:443, gae-spclient.spotify.com:443, guc-spclient.spotify.com:443]}
2020-05-10 16:40:45,916 [main] INFO  ZeroconfServer:383 - Zeroconf HTTP server started successfully on port 31867!
2020-05-10 16:40:47,575 [main] INFO  Zeroconf:456 - Announced service: Service{alias='librespot-java', service='spotify-connect', port=31867, text={VERSION=1.0, CPath=/, Stack=SP}, addresses=[/192.168.2.201], domain='.local', protocol='tcp', host='soundpi1'}
2020-05-10 16:41:05,345 [zeroconf-client-2252113] DEBUG TimeProvider:73 - Loaded time offset from NTP: 0ms
2020-05-10 16:41:12,307 [zeroconf-client-2252113] INFO  Session:122 - Created new session! {deviceId: 60a2c41c80388f39fad9e5883318693fc8d14085, ap: gae2-accesspoint-b-nfnq.ap.spotify.com:443, proxy: false} 
2020-05-10 16:41:12,313 [zeroconf-client-2252113] INFO  ZeroconfServer:343 - Accepted new user from SM-G965F. {deviceId: 60a2c41c80388f39fad9e5883318693fc8d14085}
2020-05-10 16:41:15,653 [zeroconf-client-2252113] INFO  Session:320 - Connected successfully!
2020-05-10 16:41:16,535 [cache-maintenance] INFO  CacheManager:71 - There are 7 cached entries.
2020-05-10 16:41:17,022 [zeroconf-client-2252113] INFO  LineHelper:61 - Available mixers: 'default [default]', 'sndrpihifiberry [plughw:0,0]'
2020-05-10 16:41:18,742 [session-packet-receiver] INFO  Session:1155 - Skipping SecretBlock
2020-05-10 16:41:18,748 [session-packet-receiver] INFO  Session:1127 - Received LicenseVersion: 0
2020-05-10 16:41:18,753 [session-packet-receiver] INFO  Session:1117 - Received CountryCode: DE
2020-05-10 16:41:18,771 [zeroconf-client-2252113] DEBUG TokenProvider:48 - Token expired or not suitable, requesting again. {scopes: [playlist-read], oldToken: null}
2020-05-10 16:41:19,246 [session-packet-receiver] INFO  Session:1155 - Skipping LegacyWelcome
2020-05-10 16:41:19,250 [session-packet-receiver] INFO  Session:1155 - Skipping UnknownData_AllZeros
2020-05-10 16:41:19,325 [session-packet-receiver] DEBUG Session:1131 - Received 0x10: 7BDBCF03B0C9EA783DC297BA5ADE855EE9E24E3F
2020-05-10 16:41:19,332 [session-packet-receiver] INFO  Session:1082 - Skipping unknown command {cmd: 0x75, payload: 000000}
2020-05-10 16:41:19,364 [handle-packet-28204611] DEBUG MercuryClient:209 - Couldn't dispatch Mercury event {seq: 7369874708377894912, uri: hm://pusher/v1/connections/NjBhMmM0MWM4MDM4OGYzOWZhZDllNTg4MzMxODY5M2ZjOGQxNDA4NStBUCt0Y3A6Ly9nYWUyLWFjY2Vzc3BvaW50LWItbmZucS5nYWUyLnNwb3RpZnkubmV0OjUwMjUrNjlCQjA2QjQxMDJBRThFNjFGQTE3Nzk2NDk5M0E0RDFBNzdCQ0E0MEM5QUIyNjc5QzI1N0NGMTZFMjlCMEQ5RA%3D%3D, code: 200, payload: []}
2020-05-10 16:41:19,644 [zeroconf-client-2252113] DEBUG TokenProvider:52 - Updated token successfully! {scopes: [playlist-read], newToken: StoredToken{expiresIn=3600, accessToken='BQAjv-nzr0-_8xv1GJ7dTpoQbGH-nKVHrBGL8bb3W381vWGoOFzr0iSaRkU5Tiwc3rwfTXhgbwTGOYZk5kF4K7-Dvq7UaiT983UCSwykbJQwVXT-QyOAnotjFUhoPqr8JHVdOKleIRYZtIgxC56C4scr2nY18fO4QXqw27co6OOOnZ7D0lkLL9fdlHmlS_kkQVZKzgAaIakOpaEce1B4XlVBO_Tfwru1Eg', scopes=[playlist-read], timestamp=1589128879640}}
2020-05-10 16:41:21,647 [zeroconf-client-2252113] INFO  Session:351 - Authenticated as wamvxztrfpcomxphv6401as55!
2020-05-10 16:41:25,069 [zeroconf-client-2252113] DEBUG TimeProvider:73 - Loaded time offset from NTP: 0ms
2020-05-10 16:41:25,165 [OkHttp https://guc-dealer.spotify.com/...] DEBUG DealerClient:315 - Dealer connected! {host: guc-dealer.spotify.com}
2020-05-10 16:41:25,253 [dealer-worker] DEBUG DeviceStateHandler:134 - Updated Spotify-Connection-Id: NjBhMmM0MWM4MDM4OGYzOWZhZDllNTg4MzMxODY5M2ZjOGQxNDA4NStkZWFsZXIrdGNwOi8vZ3VjMy1kZWFsZXItYi1mbTB3Lmd1YzMuc3BvdGlmeS5uZXQ6NTcwMCtDRjk2OTY4NUE4QjI3QTdENUM3NDk0OUJCQzA3RTQ3ODY4RjQ0OEFDN0U2RTYxOUVDQzkzQjUzRDY2ODc4OTRG
2020-05-10 16:41:25,458 [dealer-worker] INFO  StateWrapper:267 - Notified new device (us)!
2020-05-10 16:41:26,663 [zeroconf-client-2252113] INFO  Session:122 - Created new session! {deviceId: 60a2c41c80388f39fad9e5883318693fc8d14085, ap: gew1-accesspoint-b-psb5.ap.spotify.com:80, proxy: false} 
2020-05-10 16:41:26,668 [zeroconf-client-2252113] INFO  ZeroconfServer:343 - Accepted new user from SM-G965F. {deviceId: 60a2c41c80388f39fad9e5883318693fc8d14085}
2020-05-10 16:41:28,563 [put-state-worker] INFO  DeviceStateHandler:244 - Put state. {ts: 1589128885401, connId: NjBhMm[truncated], reason: NEW_DEVICE, request: <available for TRACE only>}
2020-05-10 16:41:28,610 [zeroconf-client-2252113] INFO  Session:320 - Connected successfully!
2020-05-10 16:41:28,723 [cache-maintenance] INFO  CacheManager:71 - There are 7 cached entries.
2020-05-10 16:41:28,839 [zeroconf-client-2252113] INFO  LineHelper:61 - Available mixers: 'default [default]', 'sndrpihifiberry [plughw:0,0]'
2020-05-10 16:41:28,929 [dealer-worker] DEBUG Player:182 - Received command: Transfer
2020-05-10 16:41:29,979 [dealer-worker] DEBUG Player:136 - Loading context (transfer), uri: spotify:user:wamvxztrfpcomxphv6401as55:collection:artist:3a9qv6NLHnsVxJUtKOMHvD
2020-05-10 16:41:30,129 [dealer-worker] DEBUG DeviceStateHandler:192 - Device is now active. {ts: 1589128890128}
2020-05-10 16:41:30,465 [zeroconf-client-2252113] DEBUG TokenProvider:48 - Token expired or not suitable, requesting again. {scopes: [playlist-read], oldToken: null}
2020-05-10 16:41:30,471 [session-packet-receiver] INFO  Session:1155 - Skipping SecretBlock
2020-05-10 16:41:30,477 [session-packet-receiver] INFO  Session:1127 - Received LicenseVersion: 0
2020-05-10 16:41:30,481 [session-packet-receiver] INFO  Session:1117 - Received CountryCode: DE
2020-05-10 16:41:30,697 [dealer-worker] DEBUG DealerClient:107 - Handled request. {key: a4804303-a506-49e9-bf04-ea8cad4460f7, result: SUCCESS}
2020-05-10 16:41:30,752 [session-packet-receiver] INFO  Session:1155 - Skipping LegacyWelcome
2020-05-10 16:41:30,756 [session-packet-receiver] INFO  Session:1155 - Skipping UnknownData_AllZeros
2020-05-10 16:41:30,771 [session-packet-receiver] DEBUG Session:1131 - Received 0x10: 693E9897323A92EC788A44CB91B6AD2B5FE86DA4
2020-05-10 16:41:30,778 [session-packet-receiver] INFO  Session:1082 - Skipping unknown command {cmd: 0x75, payload: 000000}
2020-05-10 16:41:30,782 [handle-packet-32561618] DEBUG MercuryClient:209 - Couldn't dispatch Mercury event {seq: 8782883489511899136, uri: hm://pusher/v1/connections/NjBhMmM0MWM4MDM4OGYzOWZhZDllNTg4MzMxODY5M2ZjOGQxNDA4NStBUCt0Y3A6Ly9nZXcxLWFjY2Vzc3BvaW50LWItcHNiNS5nZXcxLnNwb3RpZnkubmV0OjUwMDQrODQxMzhBRjZBRUFDMzRFMDkzNEFEQjdGMzIwQ0JDQ0RGQjI1MDU5RjI4NUQ2NDFEQTAyRTYxMEFGQTM2Qzc2MQ%3D%3D, code: 200, payload: []}
2020-05-10 16:41:30,805 [zeroconf-client-2252113] DEBUG TokenProvider:52 - Updated token successfully! {scopes: [playlist-read], newToken: StoredToken{expiresIn=3600, accessToken='BQDvSW5FCxD69gz2dbpUUaioV7SI9ukPHD57xo20cZMKPpQNwuFtEDajrBNsA_thn9g8c4UBpIngQDgvguRypqwVCH6BdsJtkLlvGN7nPQPQFCmn2T-LNOSvT_1EXyfpFSiAUyZvhN78BMTlJUTqut448yAHFNoBX7c_vOiEb3aUhnUGvcCzRG9Bm6Am7_BR5FsPtX42XaHTSiKGFFufakJtrsdxWG1y7g', scopes=[playlist-read], timestamp=1589128890802}}
2020-05-10 16:41:30,841 [zeroconf-client-2252113] INFO  Session:351 - Authenticated as wamvxztrfpcomxphv6401as55!
2020-05-10 16:41:31,234 [put-state-worker] INFO  DeviceStateHandler:244 - Put state. {ts: 1589128890608, connId: NjBhMm[truncated], reason: PLAYER_STATE_CHANGED, request: <available for TRACE only>}
2020-05-10 16:41:31,633 [put-state-worker] INFO  DeviceStateHandler:244 - Put state. {ts: 1589128890654, connId: NjBhMm[truncated], reason: PLAYER_STATE_CHANGED, request: <available for TRACE only>}
2020-05-10 16:41:31,849 [dealer-worker] DEBUG DeviceStateHandler:159 - Received cluster update at 1589128891847: <available for TRACE only>
2020-05-10 16:41:33,974 [OkHttp https://guc-dealer.spotify.com/...] DEBUG DealerClient:315 - Dealer connected! {host: guc-dealer.spotify.com}
2020-05-10 16:41:34,086 [dealer-worker] DEBUG DeviceStateHandler:134 - Updated Spotify-Connection-Id: NjBhMmM0MWM4MDM4OGYzOWZhZDllNTg4MzMxODY5M2ZjOGQxNDA4NStkZWFsZXIrdGNwOi8vZ3VjMy1kZWFsZXItYi0ya3JrLmd1YzMuc3BvdGlmeS5uZXQ6NTcwMCs0ODUwNUZGMTUzMzkzMDY2ODZFRDNCRDc0NjM1ODhCNDlERjg4MzlBN0JENTM2QzBFMTRENTE3ODM4ODQwQ0RD
2020-05-10 16:41:34,264 [dealer-worker] INFO  StateWrapper:267 - Notified new device (us)!
2020-05-10 16:41:37,496 [put-state-worker] INFO  DeviceStateHandler:244 - Put state. {ts: 1589128894094, connId: NjBhMm[truncated], reason: NEW_DEVICE, request: <available for TRACE only>}
2020-05-10 16:41:54,802 [player-runner-looper] INFO  PlayerRunner:592 - Loaded track, name: 'Fistful Of Silence', artists: 'The Glitch Mob', gid: 1008EF34374749038D76624490027E61
2020-05-10 16:41:54,826 [player-runner-looper] DEBUG CrossfadeController:84 - Loaded intervals. {start: null, end: null}
2020-05-10 16:41:56,215 [cdn-async-11712589] DEBUG Player:384 - Playback halted on retrieving chunk 10.
2020-05-10 16:41:56,795 [put-state-worker] INFO  DeviceStateHandler:244 - Put state. {ts: 1589128915672, connId: NjBhMm[truncated], reason: PLAYER_STATE_CHANGED, request: <available for TRACE only>}
2020-05-10 16:41:57,339 [put-state-worker] INFO  DeviceStateHandler:244 - Put state. {ts: 1589128916240, connId: NjBhMm[truncated], reason: PLAYER_STATE_CHANGED, request: <available for TRACE only>}
2020-05-10 16:42:07,850 [dealer-worker] DEBUG DeviceStateHandler:159 - Received cluster update at 1589128927848: <available for TRACE only>
^C2020-05-10 16:42:12,629 [Thread-0] INFO  Session:426 - Closing session. {deviceId: 60a2c41c80388f39fad9e5883318693fc8d14085} 
2020-05-10 16:42:12,677 [Thread-0] DEBUG DealerClient:303 - Did not dispatch connection invalidated: null != xyz.gianlu.librespot.dealer.DealerClient$ConnectionHolder@5ed434
2020-05-10 16:42:12,690 [Thread-0] INFO  Session:480 - Closed session. {deviceId: 60a2c41c80388f39fad9e5883318693fc8d14085} 
2020-05-10 16:42:13,071 [Thread-0] INFO  Zeroconf:483 - Unannounced service. Service{alias='librespot-java', service='spotify-connect', port=31867, text={VERSION=1.0, CPath=/, Stack=SP}, addresses=[/192.168.2.201], domain='.local', protocol='tcp', host='soundpi1'}
```

**Comment by Kucheek** [comment]
> > Also I forget to mention I am using openjdk-headless instead of the standard one, but that should not be a problem? I've been troubleshooting for hours :(
> 
> If you run out of ideas, you can always try openJDK headful. I have a similar setup, RPi 1B with Hifiberry DAC+, and when I first tried librespot-java, it also could not see my soundcard. Can't remember the exact cause, but I ended up with non-headless openjdk for some reason. There might be some additional libraries as certain articles indicate:
> 
> https://www.raspberrypi.org/forums/viewtopic.php?t=15014

@devgianlu , might be a good idea to mention that somewhere in preconditions to prevent others from the same trouble. It took me some time before I figured that out myself.

**Comment by devgianlu** [comment]
@paulbastian That is a different issue introduced by a recent change (68cd2abe97f9bd1f1aa69b2094228ee943cabeab), should be fixed with e4bef10bdcbc9e2359099230955f2f2775c962b7, not sure. Please create a new issue if it happens again.

@Kucheek Basically, you need to use the openJDK in its headful variant? I'm asking before writing something wrong on the README.

**Comment by Kucheek** [comment]
> @Kucheek Basically, you need to use the openJDK in its headful variant? I'm asking before writing something wrong on the README.

Yes, that is what helped in my case, and also for @paulbastian . Mixer was visible to librespot-java only after I replaced headless with the headful version instead.

**Comment by paulbastian** [comment]
Yes, confirmed

**Comment by paulbastian** [comment]
You close this issue, I will try again and open a new issue if the problem persists.
Are there any nightly builds or I will need to compile myself?

**Comment by devgianlu** [comment]
@paulbastian You can download the latest snapshot from [here](https://oss.sonatype.org/content/repositories/snapshots/xyz/gianlu/librespot/librespot-core/1.4.0-SNAPSHOT/) or just build the latest commit.
